tests/unit/Services/ApmCollectorServiceTest.php

```diff
@@ -0,0 +1,148 @@
+            $this->configMock
+        );
+    }
+
+    public function testStartMeasureEventWithDefaultValues()
+    {
+        $this->eventsMock->shouldReceive('dispatch')
+            ->once()
+            ->with(Mockery::on(function ($event) {
+                $this->assertEquals(self::SPAN_NAME, $event->name);
+                $this->assertEquals('request', $event->type);
+                $this->assertNull($event->action);
+                $this->assertNull($event->label);
+                $this->assertNull($event->start_time);
+
+                return true;
+            }));
+
+        $this->collectorService->startMeasure(self::SPAN_NAME);
+    }
+
+    public function testStartMeasureEventWithValues()
+    {
+        $this->eventsMock->shouldReceive('dispatch')
+            ->once()
+            ->with(Mockery::on(function ($event) {
+                $this->assertEquals(self::SPAN_NAME, $event->name);
+                $this->assertEquals('db.query', $event->type);
+                $this->assertEquals('SELECT', $event->action);
+                $this->assertEquals('Company SELECT', $event->label);
+                $this->assertEquals(1000.0, $event->start_time);
+
+                return true;
+            }));
+
+        $this->collectorService->startMeasure(
+            self::SPAN_NAME,
+            'db.query',
+            'SELECT',
+            'Company SELECT',
+            1000.0,
+        );
+    }
+
+    public function testStopMeasureEventWithDefaultValues()
+    {
+        $this->eventsMock->shouldReceive('dispatch')
+            ->once()
+            ->with(Mockery::on(function ($event) {
+                $this->assertEquals(self::SPAN_NAME, $event->name);
+                $this->assertEquals([], $event->params);
+
+                return true;
+            }));
+
+        $this->collectorService->stopMeasure(self::SPAN_NAME);
+    }
+
+    public function testStopMeasureEventWithValues()
+    {
+        $this->eventsMock->shouldReceive('dispatch')
+            ->once()
+            ->with(Mockery::on(function ($event) {
+                $this->assertEquals(self::SPAN_NAME, $event->name);
+                $this->assertEquals(['user' => 1], $event->params);
+
+                return true;
+            }));
+
+        $this->collectorService->stopMeasure(
+            self::SPAN_NAME,
+            ['user' => 1],
+        );
+    }
+
+    public function testAddCollector()
+    {
+        $agentMock = Mockery::mock(Agent::class);
+        $collectorMock = Mockery::mock(DataCollector::class);
+
+        $agentMock->shouldReceive('addCollector')
+            ->with($collectorMock);
+
+        $this->appMock->shouldReceive('make')
+            ->with(Agent::class)
+            ->andReturn($agentMock);
+
+        $this->appMock->shouldReceive('make')
+            ->with('App\Collectors\MyCollector')
+            ->andReturn($collectorMock);
+
+        $this->collectorService->addCollector('App\Collectors\MyCollector');
```

Comment:
Would this also work with a class const? `MyCollector::class`?

Reply:
Yes :)

Reply:
🤞